src/Analysis/Ast/Impl/Specializations/Typing/TypingModule.cs

```diff
@@ -227,27 +235,32 @@ private IPythonType CreateTupleType(IReadOnlyList<IPythonType> typeArgs)
             return Interpreter.UnknownType;
         }
 
-        private IPythonType CreateTypeAlias(IArgumentSet args) {
-            var typeArgs = args.Values<IMember>();
-            if (typeArgs.Count == 2) {
-                var typeName = (typeArgs[0] as IPythonConstant)?.Value as string;
-                if (!string.IsNullOrEmpty(typeName)) {
-                    return new TypeAlias(typeName, typeArgs[1].GetPythonType() ?? Interpreter.UnknownType);
-                }
+        private IPythonType CreateTypeAlias(IArgumentSet argSet) {
+            if (argSet.Errors.Count > 0) {
+                argSet.ReportErrors();
+                return Interpreter.UnknownType;
+            }
+
+            var args = argSet.Values<IMember>();
+            var name = (args[0] as IPythonConstant)?.GetString();
 
-                var firstArgType = (typeArgs[0] as PythonInstance)?.Type.Name;
-                var eval = args.Eval;
-                var expression = args.Expression;
+            if (!string.IsNullOrEmpty(name)) {
+                return new TypeAlias(name, args[1].GetPythonType() ?? Interpreter.UnknownType);
+            }
```

Comment:
need to check `args .Count > 1`

Reply:
So with the argument set change, I specified that the arguments will be name and a list of types, which means the arguments can never be null here.

Reply:
Yeah but what about _count_ of arguments.

Reply:
Number of arguments will always be 2. I specified the arguments for the first one to be `name` and the next one to be `type`, which means even if they are not provided by the user, there will still be 2 arguments. Moreover, if the user fails to provide arguments or provides too many, the earlier check with argument errors will return early. But even if we fail to detect weird cases, the number of arguments should be 2.

Reply:
Then you should add argument check on this. Like `Check.Arguments...` since the call here come from public `o.SetReturnValueProvider((interpreter, overload, args) => CreateTypeAlias(args));` and it is hard to tell that some other code somewhere will guarantee 2 args.

Reply:
OK